### PR TITLE
Tweaks to get nimbus component working with latest appservices.

### PR DESCRIPTION
**Pull request by rfk**
A few tweaks to get https://github.com/mozilla-mobile/android-components/pull/8261 working on top of the revised appservices+nimbus megazord from https://github.com/mozilla/application-services/pull/3701

### CODE REVIEW

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt [hunk 1]
```diff
@@ -102,7 +103,7 @@ open class PlacesHistoryStorage(
 
     override suspend fun getTopFrecentSites(numItems: Int): List<TopFrecentSiteInfo> {
         return withContext(readScope.coroutineContext) {
-            places.reader().getTopFrecentSiteInfos(numItems).map { it.into() }
+            places.reader().getTopFrecentSiteInfos(numItems, frecencyThreshold = FrecencyThresholdOption.NONE).map { it.into() }
```
Comment (rfk) [inline components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt:106]: This fixed compatibility with an unrelated change that's in appservices main branch. We should probably pull it out into a separate PR.

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt [hunk 2]
```diff
@@ -102,8 +103,12 @@ object Nimbus : NimbusApi {
...
         onExperimentUpdated: ((activeExperiments: List<EnrolledExperiment>) -> Unit)? = null
     ) {
-        // Set the name of the native library
-        System.setProperty("uniffi.component.nimbus.libraryOverride", "megazord")
+        // Set the name of the native library so that we use
+        // the appservices megazord for compiled code.
+        System.setProperty(
+            "uniffi.component.nimbus.libraryOverride",
+            System.getProperty("mozilla.appservices.megazord.library", "megazord")
+        )
```
Comment (rfk) [inline components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:111]: This was the simplest thing I could come up with for working with custom megazords.

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt [hunk 3]
```diff
@@ -160,7 +167,8 @@ object Nimbus : NimbusApi {
 
     override fun optOutAll() {
         if (!isInitialized) return
-        nimbus.optOutAll()
+        //nimbus.optOutAll()
+        throw RuntimeException("global opt-out hasn't landed in nimbus yet")
```
Comment (rfk) [inline components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:171]: Do we want to wait for this `optOutAll` API to land before cutting the sequence of releases?

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt (hunk 1 above)
Comment (travis79) [reply components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt:106]: Up to you on putting it into a separate PR. I don't have any strong opinion on it sneaking in here, really.

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt (hunk 2 above)
Comment (travis79) [reply components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:111]: Works for me

File: components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt (hunk 3 above)
Comment (travis79) [reply components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt:171]: After speaking with @jhugman, I think yes, let's land that first.

**Review by travis79 — APPROVED** [review]
